shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-common/src/main/java/org/apache/shardingsphere/sharding/algorithm/sharding/classbased/ClassBasedShardingAlgorithmStrategy.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.sharding.algorithm.sharding.classbased;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * Class based sharding strategy.
+ */
+@RequiredArgsConstructor
+@Getter
+public enum ClassBasedShardingAlgorithmStrategy {
```

Comment:
There are ShardingStrategy and ShardingAlgorithm in current ShardingSphere, 
So name of class ClassBasedShardingAlgorithmStrategy maybe not good enough.

Maybe `ClassBasedShardingAlgorithmType` is better, what do you think?

Reply:
I think it is not good enough too, but our props key is 'strategy', if the class named 'type', is it strange?

Reply:
How about strategy_type?